rust/src/lib.rs

```diff
@@ -329,365 +345,91 @@ pub extern "C" fn zcashlc_init_accounts_table_with_keys(
-#[no_mangle]
-pub extern "C" fn zcashlc_derive_transparent_address_from_public_key(
-    pubkey: *const c_char,
-    network_id: u32,
-) -> *mut c_char {
-    #[allow(deprecated)]
-    let res = catch_panic(|| {
-        let network = parse_network(network_id)?;
-        let public_key_str = unsafe { CStr::from_ptr(pubkey).to_str()? };
-        let pk = PublicKey::from_str(public_key_str)?;
-        let taddr = legacy::keys::pubkey_to_address(&pk).encode(&network);
-
-        Ok(CString::new(taddr).unwrap().into_raw())
-    });
-    unwrap_exc_or_null(res)
-}
-
-/// Derives a unified address from the given unified full viewing key.
-///
-/// Returns the Bech32-encoded string representation of the unified address,
-/// encoded as a null-terminated UTF-8 string.
-///
-/// # Safety
-///
-/// - `ufvk` must be non-null and must point to a null-terminated UTF-8 string representing
-///   a Bech32-encoded unified full viewing key for the given network.
-/// - The memory referenced by `ufvk` must not be mutated for the duration of the function call.
-/// - Call [`zcashlc_string_free`] to free the memory associated with the returned pointer
-///   when done using it.
-#[no_mangle]
-pub extern "C" fn zcashlc_derive_unified_address_from_viewing_key(
-    ufvk: *const c_char,
-    network_id: u32,
-) -> *mut c_char {
-    let res = catch_panic(|| {
-        let network = parse_network(network_id)?;
-        let ufvk_string = unsafe { CStr::from_ptr(ufvk).to_str()? };
-        let ufvk = match UnifiedFullViewingKey::decode(&network, ufvk_string) {
-            Ok(ufvk) => ufvk,
-            Err(e) => {
-                return Err(format_err!(
-                    "Error while deriving viewing key from string input: {}",
-                    e
-                ));
-            }
-        };
-        // Derive the default Unified Address (containing the default Sapling payment
-        // address that older SDKs used).
-        let (ua, _) = ufvk.default_address();
-        let address_str = ua.encode(&network);
-        Ok(CString::new(address_str).unwrap().into_raw())
-    });
-    unwrap_exc_or_null(res)
-}
-
-/// Derives a Sapling extended full viewing key from address from the given extended
-/// spending key.
-///
-/// Returns the Bech32-encoded string representation of the ExtendedFullViewingKey,
-/// encoded as a null-terminated UTF-8 string.
-///
-/// # Safety
-///
-/// - `extsk` must be non-null and must point to a null-terminated UTF-8 string representing
-///   a Bech32-encoded Sapling extended spending key for the given network.
-/// - The memory referenced by `extsk` must not be mutated for the duration of the function call.
-/// - Call [`zcashlc_string_free`] to free the memory associated with the returned pointer
-///   when done using it.
+///   when you are done using it.
 #[no_mangle]
-pub extern "C" fn zcashlc_derive_extended_full_viewing_key(
-    extsk: *const c_char,
+pub extern "C" fn zcashlc_spending_key_to_full_viewing_key(
```

Comment:
TODO: add derivation for IVKs, since that's something we want to expose in the SDK.

Reply:
Can we open a github issue for this so it's tracked somewhere?